robot_skills/src/robot_skills/robot.py

```diff
@@ -16,28 +18,35 @@
 
 CONNECTION_TIMEOUT = 10.0  # Timeout: all ROS connections must be alive within this duration
 
+_ = tf2_geometry_msgs
+__ = tf2_kdl
+
 
 class Robot(object):
     """
     Interface to all parts of the robot.
     """
-    def __init__(self, robot_name="", wait_services=False, tf_listener=None):
+    def __init__(self, robot_name="", wait_services=False, tf=None):
         """
         Constructor
 
         :param robot_name: Name of the robot
         :type robot_name: str
         :param wait_services: Not used anymore
         :type wait_services: bool
-        :param tf_listener: tf_listener object
-        :type tf_listener: Optional[tf.TransformListener]
+        :param tf: tf object
+        :type tf: Optional[tf2_ros.Buffer]
         """
 
         if wait_services:
             rospy.logwarn("(Robot) wait_services is not used anymore and will be removed in the future")
 
         self.robot_name = robot_name
-        self.tf_listener = tf.TransformListener() if tf_listener is None else tf_listener
+        if tf is None:
+            self.tf = tf2_ros.Buffer()
```

Comment:
Big hairy comment: why nog call this `self.tf_buffer`?

Reply:
Also fine by me.